### ISSUE TITLE
How to build using c++11

### ISSUE THREAD

**Issue by samuelstjean** [scilus/fibernavigator#222]
Or it might be a gcc 7.1 thing, not sure. Anyway, the build step throws me errors even though I tried a bunch of flags in the cmake file or the commandline such as 

CMAKE_C_FLAGS                    -std=gnu++11 

with and without the dash.

I am using wxwidgets 3.0.3.1 (the gtk2 version) on my arch laptop, so everything is at the latest version. If it's too obscure no big deal, my ubuntu 16.04 desktop works fine, I tried yesterday. If not, it's mostly before someone else also has problems and how to fix it.

~~~bash
samuel ~/git/fibernavigator/build (master) $ make CXXFLAGS="g++ -std=c++11"
[  1%] Building CXX object CMakeFiles/fibernavigator.dir/main.o
In file included from /usr/include/c++/7.1.1/type_traits:35:0,
                 from /usr/include/wx-3.0/wx/strvararg.h:25,
                 from /usr/include/wx-3.0/wx/string.h:46,
                 from /usr/include/wx-3.0/wx/memory.h:15,
                 from /usr/include/wx-3.0/wx/object.h:19,
                 from /usr/include/wx-3.0/wx/event.h:16,
                 from /usr/include/wx-3.0/wx/app.h:19,
                 from /home/samuel/git/fibernavigator/src/main.h:4,
                 from /home/samuel/git/fibernavigator/src/main.cpp:8:
/usr/include/c++/7.1.1/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^~~~~
In file included from /usr/include/wx-3.0/wx/string.h:46:0,
                 from /usr/include/wx-3.0/wx/memory.h:15,
                 from /usr/include/wx-3.0/wx/object.h:19,
                 from /usr/include/wx-3.0/wx/event.h:16,
                 from /usr/include/wx-3.0/wx/app.h:19,
                 from /home/samuel/git/fibernavigator/src/main.h:4,
                 from /home/samuel/git/fibernavigator/src/main.cpp:8:
/usr/include/wx-3.0/wx/strvararg.h:350:18: error: « is_enum » dans l'espace de noms « std » ne nomme pas un type de patron
     typedef std::is_enum<T> is_enum;
                  ^~~~~~~
/usr/include/wx-3.0/wx/strvararg.h:354:54: error: « is_enum » n'a pas été déclaré dans cette portée
     enum { value = wxFormatStringSpecifierNonPodType<is_enum::value>::value };
                                                      ^~~~~~~
/usr/include/wx-3.0/wx/strvararg.h:354:54: note: alternatives suggérées: « isalnum »
     enum { value = wxFormatStringSpecifierNonPodType<is_enum::value>::value };
                                                      ^~~~~~~
                                                      isalnum
/usr/include/wx-3.0/wx/strvararg.h:354:68: error: l'argument 1 du patron est invalide
     enum { value = wxFormatStringSpecifierNonPodType<is_enum::value>::value };
                                                                    ^
In file included from /home/samuel/git/fibernavigator/src/dataset/../misc/Fantom/FTensor.h:852:0,
                 from /home/samuel/git/fibernavigator/src/dataset/../misc/Fantom/FMatrix.icc:18,
                 from /home/samuel/git/fibernavigator/src/dataset/../misc/Fantom/FMatrix.h:515,
                 from /home/samuel/git/fibernavigator/src/dataset/ODFs.h:17,
                 from /home/samuel/git/fibernavigator/src/dataset/DatasetManager.h:7,
                 from /home/samuel/git/fibernavigator/src/main.cpp:11:
/home/samuel/git/fibernavigator/src/dataset/../misc/Fantom/FTensor.icc: Dans la fonction membre « void FTensor::resizeTensor(unsigned char, unsigned char) »:
/home/samuel/git/fibernavigator/src/dataset/../misc/Fantom/FTensor.icc:142:32: warning: la comparaison avec soi-même retourne toujours faux [-Wtautological-compare]
   if ( dimension!=dim || order != order )
                          ~~~~~~^~~~~~~~
make[2]: *** [CMakeFiles/fibernavigator.dir/build.make:63: CMakeFiles/fibernavigator.dir/main.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/fibernavigator.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
~~~

**Comment by chamberm** [comment]
Good that this is known. Isnt there something like it has to be gtk3 ?

My windows and linux versions uses wx3.0.2 also

**Comment by samuelstjean** [comment]
I think the problem is (obviously) on my end, as the wxwidget libs in archlinux are now using the c++11 ABI when they are built now.

So of course fibernav also needs to use c++11 just like the lib it uses underneath, which is easily added with a set (CMAKE_CXX_STANDARD 11) in the cmake file.

It also complained at the very end about missing webkit v1 functionality (I had webkit v2 installed), which was just an install away, so it might pull headers which are not needed, but that's not super critical I guess..

Now the questions is, would it be advantageous to change it here for future purpose or that would break every installation in existence which is not forcing/using c++11? In any case, I got it building by just forcing it over c++11, so I can add it to the wiki build instruction for the next guy in line I guess, although I just edited the cmakelist.txt manually for that.
